server/resource.go

```diff
@@ -134,11 +135,27 @@ func (s *Server) registerResourceRoutes(g *echo.Group) {
 
 			if storage.Type == api.StorageS3 {
 				s3Config := storage.Config.S3Config
+				t := time.Now()
+				s3FileKey := s3Config.Path
+				if s3Config.Path == "" {
+					s3FileKey = filename
+				} else if !strings.Contains(s3Config.Path, "{filename}") {
+					s3FileKey = path.Join(s3Config.Path, filename)
+				} else {
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{filename}", filename)
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{filetype}", filetype)
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{timestamp}", fmt.Sprintf("%d", t.Unix()))
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{year}", fmt.Sprintf("%d", t.Year()))
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{month}", fmt.Sprintf("%02d", t.Month()))
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{day}", fmt.Sprintf("%02d", t.Day()))
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{hour}", fmt.Sprintf("%02d", t.Hour()))
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{minute}", fmt.Sprintf("%02d", t.Minute()))
+					s3FileKey = strings.ReplaceAll(s3FileKey, "{second}", fmt.Sprintf("%02d", t.Second()))
+				}
 				s3client, err := s3.NewClient(ctx, &s3.Config{
 					AccessKey: s3Config.AccessKey,
 					SecretKey: s3Config.SecretKey,
 					EndPoint:  s3Config.EndPoint,
-					Path:      s3Config.Path,
```

Comment:
Why remove this line?

Reply:
https://github.com/usememos/memos/blob/main/plugin/storage/s3/s3.go#LL60C16-L60C16

Now we only need to customize the "key" since the "path" is a part of the "key".

Reply:
I see. Then we should remove the `Path` field in config https://github.com/usememos/memos/blob/main/plugin/storage/s3/s3.go#LL22C2-L22C6